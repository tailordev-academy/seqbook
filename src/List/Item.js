import React from 'react';
import PropTypes from 'prop-types';

const Item = ({ title, onSelect }) => (
  <button
    onClick={onSelect}
    className="list-group-item"
  >
    {title}
  </button>
);

Item.propTypes = {
  title: PropTypes.string.isRequired,
  onSelect: PropTypes.func.isRequired,
};

export default Item;
