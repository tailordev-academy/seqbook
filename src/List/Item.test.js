import React from 'react';
import renderer from 'react-test-renderer';

import Item from './Item';

it('renders correctly', () => {
  const tree = renderer.create(
    <Item title="item title" onSelect={() => {}} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
