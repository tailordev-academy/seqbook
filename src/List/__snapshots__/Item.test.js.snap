// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="list-group-item"
  onClick={[Function]}
>
  item title
</button>
`;
